pkg/provisioner/crdprovisioner.go

```diff
@@ -846,6 +849,10 @@ func (c *CrdProvisioner) UnpublishVolume(
 		return err
 	}
 
+	if !c.mountReplicasEnabled {
```

Comment:
MaxShares on AzVolume should already be set to 0, right? I don't think we need to check mountReplcias here.

Reply:
Consider a driver update scenario. Initially the mountReplicasEnabled was true, then we updated it to false.

Reply:
On driver upgrade the recovery function should set the correct value in the CRDs. I don't think we should keep the maxshares value and the replica management inconsistent.

Reply:
fixed